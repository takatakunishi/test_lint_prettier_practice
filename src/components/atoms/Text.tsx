type TextPropsType = {
  text: string
}

const Text: React.FC<TextPropsType> = (props) => {
  const { text } = props;
  return (
    <div>
      {text}
    </div>
  );
};

export default Text;
