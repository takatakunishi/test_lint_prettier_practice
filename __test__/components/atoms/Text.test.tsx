import { render } from '@testing-library/react'
import Text from 'components/atoms/Text'

describe('Textコンポーネントテスト', () => {
  test('日本語文字が表示されるか', () => {
    const testText = '日本語文。サンプルですゑゐ'
    basicCheckTextCode(testText)
  })
  test('英語語文字が表示されるか', () => {
    const testText = 'Hello. \n\nThis test is not difficult, but important.'
    basicCheckTextCode(testText)
  })
})

const basicCheckTextCode = (testText: string) => {
  const { container, asFragment } = render(<Text text={testText} />)
  expect(asFragment()).toMatchSnapshot()
  const innerHTMLText = container.innerHTML
  expect(innerHTMLText).toMatch(testText)
}
