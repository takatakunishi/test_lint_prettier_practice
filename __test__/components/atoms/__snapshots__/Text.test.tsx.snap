// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textコンポーネントテスト 日本語文字が表示されるか 1`] = `
<DocumentFragment>
  <div>
    日本語文。サンプルですゑゐ
  </div>
</DocumentFragment>
`;

exports[`Textコンポーネントテスト 英語語文字が表示されるか 1`] = `
<DocumentFragment>
  <div>
    Hello. 

This test is not difficult, but important.
  </div>
</DocumentFragment>
`;
